packages/playwright/src/matchers/expect.ts

```diff
@@ -289,8 +289,9 @@ class ExpectMetaInfoProxyHandler implements ProxyHandler<any> {
 
       const step = testInfo._addStep(stepInfo);
 
-      const reportStepError = (jestError: ExpectError) => {
-        const error = new ExpectError(jestError, customMessage, stackFrames);
+      const reportStepError = (_error: unknown) => {
+        const jestError = _error instanceof Error ? _error : new Error(String(_error));
```

Comment:
Why wrapping into `Error(...)`? I'd think the next line is sufficient.

Reply:
Because both `step.complete()` and `testInfo._failWithError` require the `Error` type and don't take `unknown`.

Reply:
Actually, both of these are ready for non-`Error` errors, but the types are incorrect. I think we should add a test for this just in case, similar to the one in this PR.

Reply:
Done in https://github.com/microsoft/playwright/pull/32257/commits/e4b1dc95dc5155dfbad915358cda5b09f89d4c7d. The test in this PR already has the non-`Error` case.